aten/src/ATen/templates/Tensor.h

```diff
@@ -259,35 +262,288 @@ struct CAFFE2_API Tensor {
   friend struct WeakTensor;
 
 protected:
-  c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl> tensor_impl_;
+  c10::intrusive_ptr<TensorImpl, UndefinedTensorImpl> impl_;
+
+public:
+  explicit Tensor(Storage storage)
+      : impl_(c10::make_intrusive<TensorImpl, UndefinedTensorImpl>(std::move(storage))) {}
+
+  /**
+   * @brief Creates a tensor of the given dimension.
+   *
+   * Note that the actual data allocation is not going to be carried out until
+   * the first time mutable_data() is called.
+   */
+  explicit Tensor(const std::vector<int64_t>& dims, DeviceType type)
```

Comment:
why don't these take IntLists?

Reply:
The "non-reason" is, that's just what Caffe2 had, I'm merging it :)